src/reducers/MoviesReducer.js

```diff
@@ -1,11 +1,21 @@
 export const MoviesReducer = (state = [], action) => {
-	const { type, payload } = action;
-	switch (type) {
-		case '':
-			return;
-		default:
-			return state;
-	}
+  const { type, payload } = action;
+  switch (type) {
+    case 'GET_ALL_MOVIES':
+      return [...payload.movies];
+    case 'SELECT_CATEGORY':
+      return [...payload.movies];
+    case 'TOGGLE_FAVORITE':
+      const updatedState = state.map(movie => {
+        if (movie.id === payload.id) {
+          movie.isFavorited = !movie.isFavorited;
```

Comment:
Go Team on this map function! Do we need to do anything else to push a favorite into a specific users favorite array? Or does that go somewhere else.